File: step02activity/src/main/java/com/gura/step02activity/MainActivity.java
package com.gura.step02activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity
                            implements View.OnClickListener{
    // View.OnClickListener 인터페이스 type 을 필드에 가지고 있기
    View.OnClickListener listener=new View.OnClickListener(){

        @Override
        public void onClick(View view) {
            //1. EditText 에 입력한 문자열을 읽어온다.
            EditText inputMsg=findViewById(R.id.inputMsg); // id="@+id/inputMsg"
            String msg=inputMsg.getText().toString();
            //2. Toast 메세지에 문자열을 띄우기
            Toast.makeText(MainActivity.this, msg, Toast.LENGTH_SHORT).show();
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //전송2 버튼의 참조값 얻어오기
        Button sendBtn=findViewById(R.id.sendBtn);
        //버튼에 클릭 리스너 등록하기
        sendBtn.setOnClickListener(this);

        //전송3 버튼의 참조값 얻어오기
        Button sendBtn2=findViewById(R.id.sendBtn2);
        sendBtn2.setOnClickListener(listener);
    }
    //전송 버튼을 클릭하면 호출되는 함수
    public void sendClicked(View v){
        //1. EditText 에 입력한 문자열을 읽어온다.
        EditText inputMsg=findViewById(R.id.inputMsg); // id="@+id/inputMsg"
        String msg=inputMsg.getText().toString();
        //2. Toast 메세지에 문자열을 띄우기
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
    }
    //리스너를 등록한 버튼을 클릭하면 호출되는 메소드
    @Override
    public void onClick(View view) {
        //1. EditText 에 입력한 문자열을 읽어온다.
        EditText inputMsg=findViewById(R.id.inputMsg); // id="@+id/inputMsg"
        String msg=inputMsg.getText().toString();
        //2. Toast 메세지에 문자열을 띄우기
        Toast.makeText(this, msg, Toast.LENGTH_LONG).show();
    }
    //다음 예제로 이동하기 버튼을 눌렀을때 호출되는 메소드
    public void move(View v){
        //액티비티를 이동하기 위한 Intent(의도) 객체를 생성한다.
        Intent intent=new Intent(this,
                Example2Activity.class);
        //startActivity() 메소드를 호출하면서 Intent 객체를 전달한다.
        startActivity(intent);
    }
}

File: step21hellokotlin/src/main/java/com/gura/step21hellokotlin/MainActivity.kt
package com.gura.step21hellokotlin

import android.app.AlertDialog
import android.content.DialogInterface
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.*
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener,
        AdapterView.OnItemLongClickListener{
    // null 로 property 를 초기화 하기
    //var inputMsg:EditText?=null

    // 나중에 초기화 하겠다는 keyword  lateinit
    lateinit var adapter:ArrayAdapter<String>
    lateinit var msgList:MutableList<String>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Button 에 리스너 등록
        addBtn.setOnClickListener(this)
        clearBtn.setOnClickListener(this)

        //MutableList 객체의 참조값을 얻어내서 property 에 저장하기
        msgList=mutableListOf()
        //ArrayAdapter 객체의 참조값을 얻어내서 property 에 저장하기
        adapter= ArrayAdapter(this,
                android.R.layout.simple_list_item_1,
                msgList)
        //ListView 에 아답타 연결하기
        // id 를 이용해서 View 의 참조값 바로 사용 가능
        myListView.adapter=adapter
        // 아이템 롱 클릭 리스너 등록하기
        myListView.setOnItemLongClickListener(this)

    }
    //추가, 모두 삭제 버튼을 누르면 호출되는 함수
    override fun onClick(v: View?) {
        // v 는 눌러진 버튼의 참조값이 전달된다.
        when(v?.id){
            R.id.addBtn -> {
                //입력한 문자열 읽어오기
                val msg=inputMsg.text.toString()
                Toast.makeText(this, msg, Toast.LENGTH_LONG).show()
                //입력한 문자열을 모델에 추가하고
                msgList.add(msg)
                //ListView 가 업데이트 되도록 아답타에 알린다.
                adapter.notifyDataSetChanged()
            }
            R.id.clearBtn -> {
                msgList.clear()
                adapter.notifyDataSetChanged()
            }
        }

    }

    override fun onItemLongClick(p0: AdapterView<*>?, p1: View?, position: Int, p3: Long): Boolean {
        var activity=this

        //클릭한 곳의 item
        val item:String = msgList.get(position)

        AlertDialog.Builder(this)
                .setMessage("자세히 보기로 이동 하겠습니까?")
                .setPositiveButton("네", object:DialogInterface.OnClickListener{
                    override fun onClick(p0: DialogInterface?, p1: Int) {
                        //액티비티 이동
                        //val intent=Intent(activity, DetailActivity::class.java)
                        val intent=Intent(this@MainActivity, DetailActivity::class.java)
                        //Intent 객체에 item 이라는 키값으로 String type 담기
                        intent.putExtra("item", item)
                        startActivity(intent)
                    }
                })
                .setNegativeButton("아니요", null)
                .create()
                .show();

        return false;
    }
}








